app/js/app/directives/graph_sketcher/GraphSketcher.js

```diff
@@ -1,1087 +1,317 @@
-                                        p.triangle((c.minX + c.maxX)/2 - 5, c.maxY + 2, (c.minX + c.maxX)/2 + 5, c.maxY + 2, (c.minX + c.maxX)/2, c.maxY + 7);
-                                    } else if (detect(c.minX - 3, (c.minY + c.maxY)/2)) {
-                                        p.triangle(c.minX - 2, (c.minY + c.maxY) / 2 - 5, c.minX - 2, (c.minY + c.maxY) / 2 + 5, c.minX - 7, (c.minY + c.maxY) / 2);
-                                    } else {
-                                        p.triangle(c.maxX + 2, (c.minY + c.maxY) / 2 - 5, c.maxX + 2, (c.minY + c.maxY) / 2 + 5, c.maxX + 7, (c.minY + c.maxY) / 2); 
-                                    }
-                                    p.pop();
-
-                                    found = true;
-                                    p.cursor(p.MOVE);
-                                }
-                            }
-                        }
-
-                        if (!found) {
-                            p.cursor(p.CROSS);
-                            reDraw();
-                        }
-                    }
-
-
-                    window.onkeydown = function(event) {
-                       if (event.keyCode == 46) {
-                            checkPointsUndo.push(checkPoint);
-                            checkPointsRedo = [];
-                            if (clickedCurveIdx != undefined) {
-                                var curve = (curves.splice(clickedCurveIdx, 1))[0];
-
-                                function freeAllSymbols(knots) {
-                                    for (var i = 0; i < knots.length; i++) {
-                                        var knot = knots[i];
-                                        if (knot.symbol != undefined) {
-                                            freeSymbols.push(knot.symbol);
-                                        }
-                                        if (knot.xSymbol != undefined) {
-                                            freeSymbols.push(knot.xSymbol);
-                                        }
-                                        if (knot.ySymbol != undefined) {
-                                            freeSymbols.push(knot.ySymbol);
-                                        }
-                                    }
+                        if (clickedCurveIdx != undefined) {
+                            let c = curves[clickedCurveIdx];
+                            if (mousePosition.x >= c.minX && mousePosition.x <= c.maxX && mousePosition.y >= c.minY && mousePosition.y <= c.maxY) {
+                                found = true;
+                                p.cursor(p.MOVE);
+                            } else if (detect(c.minX, c.minY) || detect(c.maxX, c.minY) || detect(c.minX, c.maxY) || detect(c.maxX, c.maxY)) {
+                                p.push();
+                                p.fill(graphViewBuilder.graphView.KNOT_DETECT_COLOR);
```

Comment:
It is me that suggested/forced accessing these values like this, it is fine for now, but I don't like it!